engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/TableSource.scala

```diff
@@ -86,9 +99,105 @@ class TableSource(
 //  TODO: add implementation during task with test from file
   override def testRecordParser: TestRecordParser[RECORD] = ???
 
+  override def generateTestData(size: Int): TestData = {
```

Comment:
It's rather complicated logic and AFAIS is unrelated to the internals of the class, so maybe we can move it to a separate trait to make this class more readable.

Reply:
True. I extracted it into a separate class - I don't see a benefit to modeling it as a trait.

As to the logic, I've extracted some methods, but its not going to be scala idiomatic since its mostly interacting with Flink's API where it just returns Unit. 

I think this class will change a lot after we add support for testing on data from the generator to accomodate reuse, but thats for another PR.